### PR TITLE
Provenance alarm

### LINKED ISSUE #2: Use the new health check API end-point for its monitoring

### PR DESCRIPTION
**Pull request by sjoshi-jpl**
## 🗒️ Summary
Update Terraform Workflow to use OIDC

## ⚙️ Test Data and/or Report
Terraform-Plan executed successfully on last push
https://github.com/NASA-PDS/monitoring/actions/runs/5072130455/jobs/9109653549

## ♻️ Related Issues

https://github.com/nasa-pds/devops/issues/45

### CODE REVIEW

File: .github/workflows/terraform-apply.yaml
Comment (alexdunnjpl) [inline .github/workflows/terraform-apply.yaml:1]: L1, L7, were these missed when copying from terraform-plan file?

**Review by alexdunnjpl — APPROVED** [review]
Code changes are approved, notwithstanding
 - potential existing errors (important to fix, assuming you don't want terraform-apply running whenever a PR is opened)
 - branch tests fail due to lint fail, request fix and re-run branch testing, suggest pre-commit hook to run linter (I have one I can send over, if that's useful)

File: .github/workflows/terraform-apply.yaml
Comment (sjoshi-jpl) [reply .github/workflows/terraform-apply.yaml:1]: @alexdunnjpl fixed most of it. The branch testing linter needs to be updated to work for terraform or we'll need to use tflint going forward. 

Will confirm this with @tloubrieu-jpl

Comment (alexdunnjpl) [reply .github/workflows/terraform-apply.yaml:1]: Awesome. Good to go.

**Review by tloubrieu-jpl — APPROVED** [review]
Thanks @sjoshi-jpl , I'll let you merge to see what happens !
Besides, we should have a documentation for the OICD configuration on our private wiki (where you had the role definition for the SAs)

**Pull request merged**